Add SageMakerProcessingSensor

vVv-AA opened the pull request:
Add SageMakerProcessingSensor which can be used to wait on a SageMaker processing job.
closes: #42411

Lee-W reviewed (commented):
In general looks good to me, but we'll need to fix the ci failure

Commit:
42411

Add SageMakerProcessingSensor which can be used to wait on a SageMaker processing job.
